src/wallet/rpcwallet.cpp

```diff
@@ -3357,7 +3399,7 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
     UniValue result(UniValue::VARR);
     std::vector<SproutNoteEntry> sproutEntries;
     std::vector<SaplingNoteEntry> saplingEntries;
-    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, fromaddress, nMinDepth, false, false);
+    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, fromaddress, nMinDepth, false, false, timestamp);
```

Comment:
If you split get filtered notes into two seperate procedures: for getting sprout and sapling notes, we could get away of constructing new vectors to hold them on each call.

This new procedures could return iterators instead of vectors, and `GetFilteredNotes` would just iterate over them and populate `sproutEntries`, `saplingEntries` vectors.
In this function you woulnd't need to call `GetFilteredNotes`.